tests/api/test_items_rest.py

```diff
@@ -1568,3 +1570,47 @@ def test_filtered_items_get(
     assert res.status_code == 200
     data = get_json(res)
     assert data['hits']['total'] == 1
+
+
+def test_items_notes(client, librarian_martigny_no_email, item_lib_martigny,
+                     json_header):
+    """Test items notes."""
+
+    item = item_lib_martigny
+    login_user_via_session(client, librarian_martigny_no_email.user)
+
+    # at start the items have no notes
+    assert len(item.notes) == 0
+
+    # add public & staff notes
+    item['notes'] = [
+        {'type': ItemNoteTypes.PUBLIC.value, 'content': 'Public note'},
+        {'type': ItemNoteTypes.STAFF.value, 'content': 'Staff note'}
+    ]
+    res = client.put(
+        url_for('invenio_records_rest.item_item', pid_value=item.pid),
+        data=json.dumps(item),
+        headers=json_header
+    )
+    assert res.status_code == 200
+
+    # add a second public note -- This should fail because we can only have one
+    # note of each type for an item
+    item['notes'].append(
+        {'type': ItemNoteTypes.PUBLIC.value, 'content': 'Second public note'}
+    )
+    with pytest.raises(RecordValidationError):
+        client.put(
+            url_for('invenio_records_rest.item_item', pid_value=item.pid),
+            data=json.dumps(item),
+            headers=json_header
+        )
+    item['notes'] = item.notes[:-1]
+
+    # get a specific type of notes
+    #  --> public : should return a note
+    #  --> checkin : should return nothing
+    #  --> dummy : should never return something !
+    assert item.get_note(ItemNoteTypes.PUBLIC.value)
+    assert item.get_note(ItemNoteTypes.CHECKIN.value) is None
+    assert item.get_note('dummy') is None
```

Comment:
See the `None` here is not clear if the field actually exists or it exist and has a value None

Reply:
Same as previous. a note content cannot be empty. So if function return `None`, this is because the item doesn't defined a note of this type.